recipes/fastjet/0001-Don-t-overwrite-external-CXXFLAGS-and-LDFLAGS.patch

```diff
@@ -0,0 +1,84 @@
+         if not OUTPUT.exists():
+-            zip_filename = DIR / pathlib.Path(CGAL_ZIP).parts[-1]
+-
+-            with urllib.request.urlopen(CGAL_ZIP) as http_obj:
+-                with open(zip_filename, "wb") as file_obj:
+-                    shutil.copyfileobj(http_obj, file_obj)
+-
+-            with zipfile.ZipFile(zip_filename) as zip_obj:
+-                cgal_dir = DIR / zip_obj.namelist()[0]
+-                zip_obj.extractall(DIR)
+-
+             # Patch for FastJet core version 3.4.0
+             # To be removed when https://gitlab.com/fastjet/fastjet/-/merge_requests/1 is merged upstream
+             subprocess.run(
+@@ -78,7 +68,8 @@ class FastJetBuild(setuptools.command.build_ext.build_ext):
+             env = os.environ.copy()
+             env["PYTHON"] = sys.executable
+             env["PYTHON_INCLUDE"] = f'-I{sysconfig.get_path("include")}'
+-            env["CXXFLAGS"] = "-O3 -Bstatic -lgmp -Bdynamic -std=c++17"
++            env["CXXFLAGS"] = env.get("CXXFLAGS", "") + " -O3 -Bstatic -Bdynamic -std=c++17"
++            env["LDFLAGS"] = env.get("LDFLAGS", "") + " -Wl,-rpath,$ORIGIN/_fastjet_core/lib"
+             env["ORIGIN"] = "$ORIGIN"  # if evaluated, it will still be '$ORIGIN'
+ 
+             args = [
+@@ -88,10 +79,9 @@ class FastJetBuild(setuptools.command.build_ext.build_ext):
+                 "--enable-allcxxplugins",
+                 "--enable-cgal-header-only",
+                 "--enable-cgal",
+-                f"--with-cgaldir={cgal_dir}",
+                 "--enable-swig",
+                 "--enable-pyext",
+-                "LDFLAGS=-Wl,-rpath,$$ORIGIN/_fastjet_core/lib:$$ORIGIN",
++                f'LDFLAGS={env["LDFLAGS"]}',
+             ]
+ 
+             try:
+@@ -106,22 +96,20 @@ class FastJetBuild(setuptools.command.build_ext.build_ext):
+                 raise
+ 
+             env = os.environ.copy()
++            env["CXX"] = env.get("CXX", "g++")
++            env["CXXFLAGS"] = env.get("CXXFLAGS", "") + " -O3 -Bstatic -Bdynamic -std=c++17"
++            env["LDFLAGS"] = env.get("LDFLAGS", "") + " -Wl,-rpath,$ORIGIN/_fastjet_core/lib"
+             env["ORIGIN"] = "$ORIGIN"  # if evaluated, it will still be '$ORIGIN'
+             subprocess.run(["make", "-j"], cwd=FASTJET, env=env, check=True)
+             subprocess.run(["make", "install"], cwd=FASTJET, env=env, check=True)
+ 
+-            subprocess.run(
+-                ["patch", "./Makefile.in", DIR / "patch_makefilein.txt"],
+-                cwd=FASTJET_CONTRIB,
+-                env=env,
+-                check=True,
+-            )
+-
+             subprocess.run(
+                 [
+                     "./configure",
+                     f"--fastjet-config={FASTJET}/fastjet-config",
+-                    "CXXFLAGS=-O3 -Bstatic -Bdynamic -std=c++17",
++                    f'CXX={env["CXX"]}',
++                    f'CXXFLAGS={env["CXXFLAGS"]}',
++                    f'LDFLAGS={env["LDFLAGS"]}',
```

Comment:
This is patching instructions to the [`v3.4.0.6` `setup.py`](https://github.com/scikit-hep/fastjet/blob/d07c735ebc5976fb8ff8ed182ab87d7b42b0354a/setup.py) to have `./configure` generate `Makefile`s for `fastjet-contrib` that contain the `CXX`, `CXXFLAGS`, and `LDFLAGS` defined here.